src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -336,27 +340,31 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr curlPtr)
 
                                 SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(chain, leafCert,
                                     checkCertName: true, hostName: easy._requestMessage.RequestUri.Host); // we disabled automatic host verification, so we do it here
-                                try
-                                {
-                                    success = serverCallback(easy._requestMessage, leafCert, chain, errors);
-                                }
-                                catch (Exception exc)
-                                {
-                                    EventSourceTrace("Server validation callback threw exception: {0}", exc, easy: easy);
```

Comment:
Just making a distinction between "you did this to yourself" and "oh, we did this to you" at the message level, instead of at the stack trace interpretation level